wordpress-mobile/WordPress-Android
# Issue/10635 material comments

Fixes #10635

This PR moves Comments List, Detail and Editor screens to Material Theme and adds Dark Theme support. There are some code changes for cosmetic reasons.

Design changes are reviewed before the code review is requested.

While refactoring is not in the scope of this PR, since I touched those files, I cleaned up/optimized some of the redundant code as suggested by AS, so do not be alarmed by the size of the PR. I suggest turning on the "Hide whitespace changes" option during the review.

There is a huge amount of PRs in this feature that covers all the screens within the app, so I'm requesting reviews from a wide variety of people.

@mattmiklic Color of PENDING status was different in List and Details, so I changed one used in the list back to yellow.

![comparison_comments](https://user-images.githubusercontent.com/728822/71392930-e8ebc180-25be-11ea-8a11-2a0e0f5c2671.png)

To test:
Navigate to Comments management screen and confirm it's visual and functional integrity in both Light and Dark themes. The theme can be changed from App Settings.

PR submission checklist:
